backend/src/main/java/com/festago/school/repository/v1/SchoolV1QueryDslRepository.java

```diff
@@ -0,0 +1,132 @@
+
+        if (response.isEmpty()) {
+            return Optional.empty();
+        }
+
+        return Optional.of(response.get(0));
+    }
+
+    public SliceResponse<SchoolFestivalV1Response> findFestivalsBySchoolId(Long schoolId, LocalDate today,
+                                                                           SchoolFestivalV1SearchCondition searchCondition) {
+        List<SchoolFestivalV1Response> result =
+            select(new QSchoolFestivalV1Response(festival.id,
+                    festival.name,
+                    festival.startDate,
+                    festival.endDate,
+                    festival.thumbnail,
+                    festivalQueryInfo.artistInfo
+                )
+            )
+                .from(festival)
+                .where(festival.school.id.eq(schoolId),
+                    addPhaseOption(searchCondition.isPast(), today),
+                    addPagingOption(searchCondition.lastFestivalId(), searchCondition.lastStartDate(),
+                        searchCondition.isPast()))
+                .leftJoin(festivalQueryInfo).on(festivalQueryInfo.festivalId.eq(festival.id))
+                .orderBy(addOrderOption(searchCondition.isPast()))
+                .limit(searchCondition.size() + NEXT_PAGE_CHECK_NUMBER)
+                .fetch();
+
+        return createResponse(searchCondition.size(), result);
+    }
+
+    private BooleanExpression addPhaseOption(boolean isPast, LocalDate today) {
+        if (isPast) {
+            return festival.endDate.lt(today);
+        }
+
+        return festival.endDate.goe(today);
+    }
+
+    private BooleanExpression addPagingOption(Long lastFestivalId, LocalDate lastStartDate, boolean isPast) {
+        if (isNotFirstPage(lastFestivalId, lastStartDate)) {
+            if (isPast) {
+                return festival.startDate.lt(lastStartDate)
+                    .or(festival.startDate.eq(lastStartDate)
+                        .and(festival.id.gt(lastFestivalId)));
+            }
+            return festival.startDate.gt(lastStartDate)
+                .or(festival.startDate.eq(lastStartDate)
+                    .and(festival.id.gt(lastFestivalId)));
+        }
+
+        return null;
+    }
+
+    private boolean isNotFirstPage(Long lastFestivalId, LocalDate lastStartDate) {
+        return lastFestivalId != null && lastStartDate != null;
+    }
+
+    private OrderSpecifier<LocalDate>[] addOrderOption(boolean isPast) {
+        if (isPast) {
+            return new OrderSpecifier[]{festival.endDate.desc()};
+        }
+
+        return new OrderSpecifier[]{festival.startDate.asc()};
+    }
+
+    private SliceResponse<SchoolFestivalV1Response> createResponse(
+        Integer pageSize,
+        List<SchoolFestivalV1Response> content) {
+        boolean isLast = true;
+        if (content.size() == pageSize + NEXT_PAGE_CHECK_NUMBER) {
+            content.remove(content.size() - NEXT_PAGE_CHECK_NUMBER);
+            isLast = false;
+        }
```

Comment:
`isLast`가 false인지 확인할 때 로직은 다음과 같이 변경할 수 있을 것 같네요!
```suggestion
        if (content.size() > pageSize) {
            content.remove(content.size() - 1);
            isLast = false;
        }
```
여기서 `content.remove(content.size() - NEXT_PAGE_CHECK_NUMBER);` 라인의 `NEXT_PAGE_CHECK_NUMBER` 상수 사용은 맞지 않은 것 같네요! (remove() 메서드는 인덱스를 지우므로)

Reply:
아하 그렇네요 만약에 2로 변경되면 하나만 삭제되겠네요